lib/CL/pocl_llvm_utils.cc

```diff
@@ -366,7 +366,7 @@ std::string CurrentWgMethod;
 
 static bool LLVMInitialized = false;
 static bool LLVMOptionsInitialized = false;
-
+static bool LLVMUseGlobalContext = true;
```

Comment:
Does the separate LLVM context per context break with the supported LLVM versions? Do we still need the global context, isn't one per cl context a better option if it works?

Reply:
It shouldn't break with any LLVM version, but it's not necessarily always better. It can lead to extreme memory usage, if the program creates lots of cl_context objects and doesn't clean them up. IIRC this was a problem with at least PyOpenCL.

Reply:
Ah right, LLVM can be large.